crates/bevy_animation/src/lib.rs

```diff
@@ -739,6 +742,143 @@ fn apply_animation(
     }
 }
 
+#[inline(always)]
+fn apply_keyframe(
```

Comment:
There's a lot of panicking indexing operations here. I understand that we probably can't get rid of them for perf reasons, but additional validation or docs on the required value of step_start would be nice.

Reply:
I added documentation for that on the `keyframes` field